### DISCUSSION TITLE
Pyinstaller doesn't work

### DISCUSSION THREAD

**Discussion by EnderiteFox** [pyinstaller/pyinstaller#6593] [Help]
I tried installing it using `pip install pyinstaller` and `py -m pip install pyinstaller`, but when I use the pyinstaller command it just says "pyinstaller is not recognised as a command", does anybody know why it happens? I tried deleting it and redownloading it 4 times. I'm running with python installed from the Microsoft store, version 3.10

**Comment by rokm** [comment]
Your python's `scripts` directory (wherever it is) is not in `PATH`. So you'll have to figure out where the scripts are installed, and add that location to `PATH` (`pip` usually shows a warning when scripts directory is not in `PATH`, so you might want to look out for that).

If you cannot figure it out, you can always invoke as a module, i..e, `python -m PyInstaller [...]` instead of `pyinstaller [...]`.

**Reply by rokm** [reply]
You can also try installing pyinstaller into a virtual environment instead of into base python. The virtual environment's scripts directory is usually added to `PATH` when you activate the venv.
